rhmap-test-driver/app/src/test-runner.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const API = require("./rhmap-api");
+
+class TestRunner {
+
+    constructor() {
+        // The ID of the application that owns the target aliases.
+        this.appId;
+        // The time between one request and another
+        this.delay;
+    }
+
+    /**
+     * The URL of the node backend
+     */
+    set endPoint(endPoint) {
+        if (endPoint.lastIndexOf('/') === endPoint.length - 1) {
+            endPoint.slice(-1);
+        }
+        this.api = new API(endPoint);
+    }
+
+    /**
+     * It sends a request to the backend endpoint for each alias with a fix delay between each one
+     */
+    start(aliases) {
+        if (!this.api || !this.appId) {
+            throw new Error("An endpoint URL and an application ID must be provided");
+        }
+
+        const message = "Hello from rhmap-test-driver!";
+
+        aliases.forEach((alias, i) => {
+            setTimeout(() => {
+                console.log(`Sending notification to ${alias}`);
+                this.api.sendNotificationToAlias(message, this.appId, alias)
+                    .on("error", err => {
+                        console.log(`[${alias}] FAILED: ${err}`)
+                    })
+                    .on("response", res => {
+                        console.log(`[${alias}] SUCCESS`)
+                    });;
+            }, this.delay * i);
```

Comment:
This piece of code is weird. It is to send a request for each alias, asynchronously and exactly with an interval of `this.delay * i`. If I use `async.eachSeries` it would not matter if I put a `sleep` inside the loop because each iteration would be async. The problem here is the fix interval.